Fabric.Identity.API/scripts/Install-Identity-Utilities.psm1

```diff
@@ -0,0 +1,621 @@
+            throw
+        }
+        try{
+            Remove-Item $env:Temp\bundle.exe
+        }catch{
+            $e = $_.Exception
+            Write-DosMessage -Level "Warning" -Message "Unable to remove temporary dowload file for server hosting bundle exe" 
+            Write-DosMessage -Level "Warning" -Message  $e.Message
+        }
+
+    }else{
+        Write-DosMessage -Level "Information" -Message  ".NET Core Windows Server Hosting Bundle (v$version) installed and meets expectations."
+    }
+}
+
+function Get-IISWebSiteForInstall([string] $selectedSiteName, [bool] $quiet){
+    try{
+        $sites = Get-ChildItem IIS:\Sites
+        if($quiet -eq $true){
+            $selectedSite = $sites | Where-Object { $_.Name -eq $selectedSiteName }
+        }else{
+            if($sites -is [array]){
+                $sites |
+                    ForEach-Object {New-Object PSCustomObject -Property @{
+                        'Id'=$_.id;
+                        'Name'=$_.name;
+                        'Physical Path'=[System.Environment]::ExpandEnvironmentVariables($_.physicalPath);
+                        'Bindings'=$_.bindings;
+                    };} |
+                    Format-Table Id,Name,'Physical Path',Bindings -AutoSize | Out-Host
+
+                $selectedSiteId = Read-Host "Select a web site by Id"
+                $selectedSite = $sites[$selectedSiteId - 1]
+            }else{
+                $selectedSite = $sites
+            }
+        }
+        if($null -eq $selectedSite){
+            throw "Could not find selected site."
+        }
+        if($selectedSite.Name){ Add-InstallationSetting "identity" "siteName" $selectedSite.Name | Out-Null }
+
+        return $selectedSite
+
+    }catch{
+        Write-DosMessage -Level "Error" -Message "Could not select a website."
+        throw
+    }
+}
+
+function Get-Certificates([string] $primarySigningCertificateThumbprint, [string] $encryptionCertificateThumbprint, [bool] $quiet){
+    if(Test-ShouldShowCertMenu -primarySigningCertificateThumbprint $primarySigningCertificateThumbprint `
+                                -encryptionCertificateThumbprint $encryptionCertificateThumbprint `
+                                -quiet $quiet){
+        try{
+            $allCerts = Get-CertsFromLocation Cert:\LocalMachine\My
+            $index = 1
+            $allCerts |
+                ForEach-Object {New-Object PSCustomObject -Property @{
+                'Index'=$index;
+                'Subject'= $_.Subject; 
+                'Name' = $_.FriendlyName; 
+                'Thumbprint' = $_.Thumbprint; 
+                'Expiration' = $_.NotAfter
+                };
+                $index ++} |
+                Format-Table Index,Name,Subject,Expiration,Thumbprint  -AutoSize | Out-Host
+    
+            $selectionNumber = Read-Host  "Select a signing and encryption certificate by Index"
```

Comment:
instead of throwing an error why not have a while loop until they provide a value

Reply:
Good idea, I'll make that change.